packages/flutter/lib/src/material/chip_theme.dart

```diff
@@ -0,0 +1,425 @@
+    final Color disabledColor = baseColor.withAlpha(disabledAlpha);
+    final Color selectedColor = baseColor.withAlpha(selectAlpha);
+    final Color secondarySelectedColor = primaryColor.withAlpha(selectAlpha);
+    final TextStyle secondaryLabelStyle = labelStyle.copyWith(
+      color: primaryColor.withAlpha(textLabelAlpha),
+    );
+    labelStyle = labelStyle.copyWith(color: baseColor.withAlpha(textLabelAlpha));
+
+    return new ChipThemeData(
+      backgroundColor: backgroundColor,
+      deleteIconColor: deleteIconColor,
+      disabledColor: disabledColor,
+      selectedColor: selectedColor,
+      secondarySelectedColor: secondarySelectedColor,
+      labelPadding: labelPadding,
+      padding: padding,
+      shape: shape,
+      labelStyle: labelStyle,
+      secondaryLabelStyle: secondaryLabelStyle,
+      brightness: brightness,
+    );
+  }
+
+  /// Color to be used for the unselected, enabled chip's background.
+  ///
+  /// The default is light grey.
+  final Color backgroundColor;
+
+  /// The [Color] for the delete icon. The default is Color(0xde000000)
+  /// (slightly transparent black) for light themes, and Color(0xdeffffff)
+  /// (slightly transparent white) for dark themes.
+  ///
+  /// May be set to null, in which case the ambient [IconTheme.color] is used.
+  final Color deleteIconColor;
+
+  /// Color to be used for the chip's background indicating that it is disabled.
+  ///
+  /// The chip is disabled when [isEnabled] is false, or all three of
+  /// [SelectableChipAttributes.onSelected], [TappableChipAttributes.onPressed],
+  /// and [DeletableChipAttributes.onDelete] are null.
+  ///
+  /// It defaults to [Colors.black38].
+  final Color disabledColor;
+
+  /// Color to be used for the chip's background, indicating that it is
+  /// selected.
+  ///
+  /// The chip is selected when [selected] is true.
+  final Color selectedColor;
+
+  /// An alternate color to be used for the chip's background, indicating that
+  /// it is selected. For example, this color is used by [ChoiceChip] when the
+  /// choice is selected.
+  ///
+  /// The chip is selected when [selected] is true.
+  final Color secondarySelectedColor;
+
+  /// The padding around the [label] widget.
+  ///
+  /// By default, this is 4 logical pixels at the beginning and the end of the
+  /// label, and zero on top and bottom.
+  final EdgeInsetsGeometry labelPadding;
+
+  /// The padding between the contents of the chip and the outside [border].
+  ///
+  /// Defaults to 4 logical pixels on all sides.
+  final EdgeInsetsGeometry padding;
+
+  /// The border to draw around the chip.
+  ///
+  /// Defaults to a [StadiumBorder]. Must not be null.
+  final ShapeBorder shape;
+
+  /// The style to be applied to the chip's label.
+  ///
+  /// This only has an effect on label widgets that respect the
+  /// [DefaultTextStyle], such as [Text].
+  final TextStyle labelStyle;
+
+  /// An alternate style to be applied to the chip's label. For example, this
```

Comment:
If this is only used to show selection, maybe it should be called `selectedLabelStyle`